brayns/network/websocket/WebSocket.cpp

```diff
@@ -80,36 +105,61 @@ class WebSocketSender
         auto flags = packet.getFlags();
         try
         {
-            socket.sendFrame(buffer, int(size), flags);
+            socket.sendFrame(buffer, static_cast<int>(size), flags);
         }
         catch (const Poco::Exception &e)
         {
             throw brayns::ConnectionClosedException("Error while sending packet: '" + e.displayText() + "'");
         }
     }
 };
+
+class SendLogger
+{
+public:
+    static void log(const brayns::OutputPacket &packet, size_t id)
+    {
+        if (packet.isBinary())
+        {
+            _logBinary(packet, id);
+            return;
+        }
+        if (packet.isText())
+        {
+            _logText(packet, id);
+            return;
+        }
+        brayns::Log::error("Trying to send invalid packet (neither text nor binary).");
+    }
+
+private:
+    static void _logBinary(const brayns::OutputPacket &packet, size_t id)
+    {
+        auto data = packet.getData();
+        auto size = data.size();
+        brayns::Log::debug("Sending binary frame of {} bytes to client {}.", size, id);
+    }
+
+    static void _logText(const brayns::OutputPacket &packet, size_t id)
+    {
+        auto data = packet.getData();
+        brayns::Log::debug("Sending text frame to client {}: '{}'.", id, data);
+    }
+};
 } // namespace
 
 namespace brayns
 {
-ConnectionClosedException::ConnectionClosedException(const std::string &message)
-    : std::runtime_error(message)
+WebSocket::WebSocket(const Poco::Net::WebSocket &socket)
+    : _socket(socket)
 {
+    _socket.setReceiveTimeout(Poco::Timespan());
+    _socket.setSendTimeout(Poco::Timespan());
 }
 
-WebSocket::WebSocket(
-    Poco::Net::HTTPClientSession &session,
-    Poco::Net::HTTPRequest &request,
-    Poco::Net::HTTPResponse &response)
-    : _socket(session, request, response)
+size_t WebSocket::getId() const
 {
-    WebSocketBuilder::build(_socket);
-}
-
-WebSocket::WebSocket(Poco::Net::HTTPServerRequest &request, Poco::Net::HTTPServerResponse &response)
-    : _socket(request, response)
-{
-    WebSocketBuilder::build(_socket);
+    return reinterpret_cast<size_t>(this);
```

Comment:
std::uintptr_t ?

Reply:
Well using the address as the ID is an implementation detail, from the caller point of view, it is supposed to be an unsigned integer and I used size_t so we can use it as an index.

In any case they both typedef to the same underlying type unless we have weird old segmented architecture.